src/parsers/manifestjson.js

```diff
@@ -468,6 +468,13 @@ export default class ManifestJSONParser extends JSONParser {
       });
     }
 
+    if (this.parsedJSON.applications?.gecko_android) {
+      this.collector.addError(
+        messages.manifestFieldUnsupported('/applications/gecko_android')
+      );
+      this.isValid = false;
+    }
+
     if (this.parsedJSON.manifest_version < 3) {
       if (
         this.parsedJSON.browser_specific_settings &&
```

Comment:
@willdurand I'm wondering if we should tweak also this check to make sure we will report the `IGNORE_APPLICATION_PROPERTY` message only when both `browser_specific_settings` and `applications` have a `gecko` property and `APPLICATIONS_DEPRECATED` otherwise, so that on a manifest.json like the following we would be using the latter instead of the former:

```
  "applications": {
    "gecko": { "id": "test@extension" }
  },
  "browser_specific_settings": {
    "gecko_android": { ... }
  }
```

wdyt?

if we agree that it would be reasonable, then we would likely be tweaking this to something like:
```
     if (
        this.parsedJSON.browser_specific_settings?.gecko &&
        this.parsedJSON.applications?.gecko
      ) {
       ...  
```

and we would need to also tweak the test case `it('only emits a single warning when a MV2 extension uses both applications and browser_specific_settings', ...)` from `tests/unit/parsers/test.manifestjson.js`.